plugins.d/python.d.plugin

```diff
@@ -0,0 +1,328 @@
+                  self.configs +
+                  mod.__name__ +
+                  ".conf' not found. Using defaults.")
+        elif reason[:11] == "misbehaving":
+            debug(mod.__name__, "is misbeaving. Disabling it")
+
+    def check(self):
+        for mod in self.modules:
+            try:
+                if not mod.check():
+                    self.disable_module(mod, "failed check")
+            except AttributeError:
+                self.disable_module(mod, "no check")
+            except UnboundLocalError:
+                self.disable_module(mod, "misbehaving")
+
+    def create(self):
+        for mod in self.modules:
+            try:
+                if not mod.create():
+                    self.disable_module(mod, "failed create")
+                else:
+                    chart = mod.__name__
+                    sys.stdout.write(
+                        "CHART netdata.plugin_pythond_" +
+                        chart +
+                        " '' 'Execution time for " +
+                        chart +
+                        " plugin' 'milliseconds / run' python.d netdata.plugin_python area 145000 " +
+                        str(self.timetable[mod.__name__][1]) +
+                        '\n')
+                    sys.stdout.write("DIMENSION run_time 'run time' absolute 1 1\n\n")
+                    sys.stdout.flush()
+            except AttributeError:
+                self.disable_module(mod, "no create")
+            except UnboundLocalError:
+                self.disable_module(mod, "misbehaving")
+
+    def update_module(self, mod):
+        t1 = time.time()
+        # check if it is time to execute module update() function
+        if (t1 - self.timetable[mod.__name__][0]) < self.timetable[mod.__name__][1]:
+            return
+        try:
+            if not self.first_run:
+                t = ""
+            else:
+                t=int((t1- self.timetable[mod.__name__][0]) * 1000000)
+            if not mod.update(t):
+                self.disable_module(mod, "update failed")
+                return
+        except AttributeError:
+            self.disable_module(mod, "no update")
+            return
+        except UnboundLocalError:
+            self.disable_module(mod, "misbehaving")
+            return
+        t2 = time.time()
+        if self.first_run:
+            dt = ""
+        else:
+            dt = int((t2 - self.timetable[mod.__name__][0]) * 1000000)
+        sys.stdout.write("BEGIN netdata.plugin_pythond_"+mod.__name__+" "+str(dt)+'\n')
+        sys.stdout.write("SET run_time = " + str(int((t2 - t1) * 1000)) + '\n')
+        sys.stdout.write("END\n")
+        sys.stdout.flush()
+        self.timetable[mod.__name__][0] = t1
+
+    def update(self):
+        while True:
+            t_begin = time.time()
+            for mod in self.modules:
+                self.update_module(mod)
+            time.sleep((self.interval - (time.time() - t_begin)) / 2)
```

Comment:
If you keep the `next_run` for every module, you should sleep for `sleep_time = min(next_run of all modules) - time.time()`. If it is zero or negative, you should not sleep.